database/models/user.model.ts

```diff
@@ -26,6 +40,44 @@ const UserSchema = new Schema(
       type: String,
       required: false,
     },
+    learningGoals: {
+      type: [String],
+      required: false,
+    },
+    role: {
+      type: [Schema.Types.ObjectId],
+      required: false,
+      default: "user",
+      ref: "Role",
+    },
+    technologyStack: {
+      type: [Schema.Types.ObjectId],
+      required: false,
+      ref: "Technology",
+    },
+    experienceLevel: {
+      type: [Schema.Types.ObjectId],
+      required: false,
+    },
+    availability: {
+      type: [Date],
+      required: false,
+    },
+    tags: {
+      type: [Schema.Types.ObjectId],
+      ref: "Tag",
+      required: false,
+    },
+    socials: {
+      type: [Schema.Types.ObjectId],
+      required: false,
+      ref: "Social",
+    },
```

Comment:
Keep in mind we're using mongo and the document model doesn't need to be treated like a regular SQL database. We can embed data with "subdocuments" when the data isn't going to ever be used independently elsewhere.

A good use-case might be the `socials` here. We're unlikely to search all the socials or want to list or edit them independently of the user profile.

```js
const SocialsSchema = new Schema({
  twitter: {
    type: String,
    required: false,
  },
  facebook: {
    type: String,
    required: false,
  },
  linkedin: {
    type: String,
    required: false,
  },
});

const UserSchema = new Schema(
  {
  // etc...
 socials: SocialsSchema,
  
```

So something like this might make more sense instead of referencing by ID. This way we don't need to make a more complex query every time we view the profile just to list this user's socials.


This is just a different way of doing things- it isn't a requirement.

Reply:
ok , noted.  I like this method, didn't know you could do that